ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -381,10 +387,36 @@ internal constructor(
   fun isLastPosition(taskId: String? = null): Boolean =
     (taskId ?: currentTaskId.value) == getTaskSequence().last().id
 
+  /**
+   * Returns true if the given [taskId] and task data would be last, or the current active task.
+   * Useful for handling conditional tasks.
+   */
+  fun testLastPosition(taskId: String? = null, value: TaskData?): Boolean =
```

Comment:
Can we call this method `isLastPosition` can omit `test` from the method name?

Reply:
Yes, looks like this method's name needs to be clarified, as per @anandwana001's comment too!

So we essentially have two methods, `isLastPosition()`, which checks that the task is truly the last in the sequence, and I renamed this one to `checkLastPositionWithTaskData()`. It would also be trivial to combine these together into one, but I'd prefer this weird edge case thing to be in its own method so that someone can investigate its origin later. I left the bug number as a bread crumb.